src/XMakeTasks/AssemblyDependency/AssemblyFoldersFromConfig/AssemblyFoldersFromConfigResolver.cs

```diff
@@ -0,0 +1,297 @@
+using Microsoft.Build.Shared;
+using Microsoft.Build.Framework;
+using ProcessorArchitecture = System.Reflection.ProcessorArchitecture;
+
+namespace Microsoft.Build.Tasks.AssemblyFoldersFromConfig
+{
+    /// <summary>
+    /// Resolve search path type {AssemblyFolderFromConfig: *}
+    /// </summary>
+    internal class AssemblyFoldersFromConfigResolver : Resolver
+    {
+        /// <summary>
+        /// Regex for breaking up the search path pieces.
+        /// </summary>
+        private static readonly Regex s_crackAssemblyFoldersFromConfigSentinel = new Regex
+        (
+            AssemblyResolutionConstants.assemblyFoldersFromConfigSentinel + "(?<ASSEMBLYFOLDERCONFIGFILE>[^,]*),(?<TARGETRUNTIMEVERSION>[^,]*)}",
+            RegexOptions.IgnoreCase
+        );
+
+        /// <summary>
+        /// Whether or not the search path could be cracked.
+        /// </summary>
+        private bool _wasMatch;
+
+        /// <summary>
+        /// From the search path.
+        /// </summary>
+        private string _targetRuntimeVersion;
+
+        /// <summary>
+        /// Whether regex initialization has happened.
+        /// </summary>
+        private bool _isInitialized; // is initialized to false automatically
+
+        /// <summary>
+        /// List of assembly folders to search for keys in.
+        /// </summary>
+        private AssemblyFoldersFromConfigCache _assemblyFoldersCache;
+
+        /// <summary>
+        /// BuildEngine
+        /// </summary>
+        private readonly IBuildEngine4 _buildEngine;
+
+        /// <summary>
+        /// Path to the assembly folder config file.
+        /// </summary>
+        private string _assemblyFolderConfigFile;
+
+        /// <summary>
+        /// If it is not initialized then just return the null object, that would mean the resolver was not called.
+        /// </summary>
+        internal AssemblyFoldersFromConfig AssemblyFoldersExLocations => _assemblyFoldersCache?.AssemblyFoldersFromConfig;
+
+        /// <summary>
+        /// Construct.
+        /// </summary>
+        public AssemblyFoldersFromConfigResolver(string searchPathElement, GetAssemblyName getAssemblyName,
+            FileExists fileExists, GetAssemblyRuntimeVersion getRuntimeVersion, Version targetedRuntimeVesion,
+            ProcessorArchitecture targetProcessorArchitecture, bool compareProcessorArchitecture,
+            IBuildEngine buildEngine)
+            : base(
+                searchPathElement, getAssemblyName, fileExists, getRuntimeVersion, targetedRuntimeVesion,
+                targetProcessorArchitecture, compareProcessorArchitecture)
+        {
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILDMYDEBUG")))
+            {
+                while (!System.Diagnostics.Debugger.IsAttached)
+                {
+                    System.Threading.Thread.Sleep(500);
+
+                }
+                System.Diagnostics.Debugger.Break();
+            }
```

Comment:
If this will continue to be useful, give it a more descriptive name than `MY`. But probably just remove it?